### PR TITLE
Don't use node globals in browser context

### PR DESCRIPTION
**Pull request by swissspidy**
## Context

Splitting this out from #5792, see #6304 for full context

## Summary

Replace usage of `global.*` with `window.*`

webpack v5 no longer polyfills Node.js APIs in the browser, including `global`, which is a Node thing, but webpack has happily supported in the past by making it an alias of `window`. Example: `global.navigator` becomes `window.navigator`.

Replace usage of `process.env` with custom variables

webpack v5 no longer polyfills Node.js APIs in the browser, including `process`. Uses variables like `WEB_STORIES_ENV` instead of `process.env.NODE_ENV` in all frontend code.

## Relevant Technical Choices

See above

## User-facing changes

None

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Partially addresses #6304

### CODE REVIEW

File: packages/wp-story-editor/src/components/mediaUpload/mediaPicker/WordPressImageCropper.js [hunk 1]
```diff
@@ -53,7 +53,7 @@ const WordPressImageCropper = window.wp?.media?.controller?.Cropper?.extend?.({
         : control.params.height;
     }
 
-    return wp.ajax.post('crop-image', {
+    return window.wp.ajax.post('crop-image', {
```
Comment (spacedmonkey) [inline packages/wp-story-editor/src/components/mediaUpload/mediaPicker/WordPressImageCropper.js:56]: 🤔 Is this required or nice to have?

Comment (spacedmonkey) [inline packages/wp-story-editor/src/components/mediaUpload/mediaPicker/WordPressImageCropper.js:56]: 🤔 Is this required or nice to have?

Comment (swissspidy) [reply packages/wp-story-editor/src/components/mediaUpload/mediaPicker/WordPressImageCropper.js:56]: It is required. I removed `wp` as a recognized global variable in the ESLint config, so that we are forced to explicitly write the full variable with `window.wp`.

Comment (swissspidy) [reply packages/wp-story-editor/src/components/mediaUpload/mediaPicker/WordPressImageCropper.js:56]: Yes. Closing in favor of the [other open thread](https://github.com/GoogleForCreators/web-stories-wp/pull/10487#discussion_r800666513)

**Review by spacedmonkey — APPROVED** [review]

File: packages/e2e-tests/src/config/bootstrap.js [hunk 2]
```diff
@@ -116,7 +116,7 @@ const pageEvents = [];
 jest.setTimeout(PUPPETEER_TIMEOUT || 100000);
 
 // Retry flaky tests at most 2 times in CI (off by 1).
-if ('true' === process.env.CI) {
+if ('true' === process.env.NODE_ENV) {
```
Comment (samwhale) [inline packages/e2e-tests/src/config/bootstrap.js]: 💭  should this be `WEB_STORIES_ENV`?

Maybe not since this is for e2e tests

File: packages/e2e-tests/src/jest.config.js [hunk 3]
```diff
@@ -37,7 +37,7 @@ export default {
   testMatch: ['**/specs/**/*.[jt]s', '**/?(*.)spec.[jt]s'],
   // @jest/test-sequencer is the default.
   testSequencer:
-    'true' === process.env.CI
+    'true' === process.env.NODE_ENV
```
Comment (samwhale) [inline packages/e2e-tests/src/jest.config.js]: Same here? I guess this one doesn't matter since process should be available when running jest.

**Review by samwhale — APPROVED** [review]

File: packages/e2e-tests/src/config/bootstrap.js (hunk 2 above)
Comment (swissspidy) [reply packages/e2e-tests/src/config/bootstrap.js]: This code is executed in Node context, not browser context. So using `process.env` is correct.

But I shouldn't have changed this to `NODE_ENV`, it should stay `process.env.CI` 🤦

File: packages/e2e-tests/src/jest.config.js (hunk 3 above)
Comment (swissspidy) [reply packages/e2e-tests/src/jest.config.js]: Yes, same here. This code is executed in Node context, not browser context. So using `process.env` is correct. 

But it really should stay `process.env.CI`
